src/integrations/pulsar/consumer.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+#include <atomic>
+#include <optional>
+#include <span>
+#include <thread>
+
+#include <pulsar/Client.h>
+
+namespace integrations::pulsar {
+
+namespace pulsar_client = ::pulsar;
+
+class Consumer;
+
+class Message final {
+ public:
+  explicit Message(pulsar_client::Message &&message);
+
+  std::span<const char> Payload() const;
+
```

Comment:
I think offset management wouldn't make sense in the context of cursors but what about subscription types? Shall we expose these to Consumers?

Reply:
It does make sense to expose something like that. But as I said to Benjo, I would leave those things for later once we have the core E2E merged (this PR + the query one). Then we can discuss what to expose for procedures and for the configs.